docs/source/configuration.rst

```diff
@@ -0,0 +1,151 @@
+located in your home directory.  The file is written in the YAML format,
+which allows for a human-readable hierarchical key-value configuration.
+All keys in the configuration file are optional, though Dask will create
+a default configuration file for you on its first launch.
+
+Here is a synopsis of the configuration file:
+
+.. code-block:: yaml
+
+   logging:
+       distributed: info
+       distributed.client: warning
+       bokeh: critical
+
+   # Scheduler options
+   bandwidth: 100000000    # 100 MB/s estimated worker-worker bandwidth
+   allowed-failures: 3     # number of retries before a task is considered bad
+   pdb-on-err: False       # enter debug mode on scheduling error
+   transition-log-length: 100000
+
+   # Worker options
+   multiprocessing-method: forkserver
+
+   # Communication options
+   compression: auto
+   tcp-timeout: 30         # seconds delay before calling an unresponsive connection dead
+   default-scheme: tcp
+   require-encryption: False   # whether to require encryption on non-local comms
+   tls:
+       ca-file: myca.pem
+       scheduler:
+           cert: mycert.pem
+           key: mykey.pem
+       worker:
+           cert: mycert.pem
+           key: mykey.pem
+       client:
+           cert: mycert.pem
+           key: mykey.pem
+       #ciphers:
+           #ECDHE-ECDSA-AES128-GCM-SHA256
+
+   # Bokeh web dashboard
+   bokeh-export-tool: False
+
+
+We will review some of those options hereafter.
+
+
+Communication options
+---------------------
+
+``compression``
+"""""""""""""""
+
+This key configures the desired compression scheme when transferring data
+over the cluster.  The default value, "auto", applies heuristics to try and
+select the best compression scheme for each piece of data.
+
+
+``default-scheme``
+""""""""""""""""""
+
+The :ref:`communication <communications>` scheme used by default.  You can
+override the default ("tcp") here, but it is recommended to use explicit URIs
+for the various endpoints instead (for example ``tls://`` if you want to
+enable :ref:`TLS <tls>` communications).
+
+
+``require-encryption``
+""""""""""""""""""""""
+
+Whether to require that all non-local communications be encrypted.  If true,
+then Dask will refuse establishing any clear-text communications (for example
+over TCP without TLS), forcing you to use a secure transport such as
+:ref:`TLS <tls>`.
+
+
+``tcp-timeout``
+"""""""""""""""
+
+The default "timeout" on TCP sockets.  If a remote endpoint is unresponsive
+(at the TCP layer, not at the distributed layer) for at least this seconds,
+the communication is considered closed.  This helps detect endpoints that
+have been killed or disconnected abruptly.
+
+
+``tls``
+"""""""
+
+This key configures :ref:`TLS <tls>` communications.  Several sub-keys are
+recognized:
+
+* ``ca-file`` configures the CA certificate file used to authenticate
+  and authorize all endpoints
+* ``ciphers`` restricts allowed ciphers on TLS communications
```

Comment:
Perhaps mention the ciphers are in OpenSSL format and their interpreted order is that of OpenSSL (I assume it is?!)?